checkr/src/pg.rs

```diff
@@ -145,23 +145,25 @@ impl Commands {
 fn guard_edges(det: Determinism, guards: &[Guard], s: Node, t: Node) -> Vec<Edge> {
```

Comment:
Aha! I think this is the primary culprit! In the definition, the non-deterministic only returns the edges, while in the deterministic it also returns the done condition.
```suggestion
/// Computes the edges and the condition which is true iff all guards are false
fn guard_edges(det: Determinism, guards: &[Guard], s: Node, t: Node) -> (Vec<Edge>, BExpr) {
```

I think the best approach here would be to return "done" in both cases, where the deterministic case would return `prev` as per the definition but with an addition added, while the non-deterministic would return `done(guards)`.

In that way, the `Loop` case on line 191 could use the same `d` returned by `guard_edges` in both cases!

Reply:
Good point, I'll look into implementing this feature in the pull request 😄